src/components/Button/Button.tsx

```diff
@@ -1,12 +1,80 @@
 import React from "react";
+import { Icon, IconChoice } from "../Icon/Icon";
 
-// Props
-type Props = {} & JSX.IntrinsicElements["button"];
+type IntrinsicElements = JSX.IntrinsicElements["button"];
 
-export const Button: React.FC<Props> = ({ children, ...rest }) => {
+interface Props extends IntrinsicElements {
+  buttonText: string;
+  buttonVariation?: ButtonVariation;
+  disabled?: boolean;
+  iconName?: IconChoice;
+  shiftIconLeft?: boolean;
+  ariaLabel?: string;
+  largeButton?: boolean;
+}
+
+/**
+ * Button Component
+ * @param {string}      buttonText            Renders the text contained in the button.
+ * @param {string}      [buttonVariation]     Renders the style of the button.
+ * @param {boolean}     [disabled]            Determines whether or not a button is enabled.
+ * @param {IconChoice}  [iconName]            Handles the icon to render in the button.
+ * @param {boolean}     [shiftIconLeft]       Determines if a rendered icon should render on the left side.
+ * @param {string}      [ariaLabel]           Overwrites the aria label of the button element that will be read on screen readers.
+ * @param {boolean}     [largeButton]         Determines whether a large button is required.
+ * @param {string}      [button]              Type of specified button.
+ * @param {Event}       [onClick]             Handles its behavior when the button is clicked.
+ * @param {string}      [target]              Specifies a name or a keyword that indicates where to display the response that is received after clicking the button.
+ */
+
+type ButtonVariation =
+  | "primary"
+  | "secondary"
+  | "error"
+  | "success"
+  | "inverse"
+  | "base"
+  | "link"
+  | "outline";
+
+const ButtonVariationConversion: { [key: string]: string } = {
+  primary: "",
+  secondary: "outline",
+  error: "secondary",
+  success: " bg-green",
+  inverse: "outline text-gray-70 bg-base-lighter",
+  base: "base",
+  link: "unstyled",
+  outline: "outline text-black",
+};
+
+export const Button: React.FC<Props> = ({
+  buttonVariation = "primary",
+  buttonText,
+  shiftIconLeft = false,
+  iconName,
+  ariaLabel,
+  largeButton = false,
+  disabled = false,
+  ...rest
+}) => {
+  const buttonVariationType = ButtonVariationConversion[buttonVariation] ?? "";
   return (
-    <button {...rest} className="usa-button">
-      {children}
+    <button
+      disabled={disabled}
+      {...rest}
+      aria-label={ariaLabel || `${buttonText} button`}
+      className={`display-flex usa-button usa-button--${buttonVariationType} ${
+        largeButton ? "usa-button--big" : ""
+      }`}
+    >
+      {shiftIconLeft && iconName && <Icon name={iconName} />}
+      <span className="padding-top-05 padding-x-05">{buttonText}</span>
```

Comment:
```suggestion
      <span className="padding-x-05">{buttonText}</span>
```
This is causing the text to align slightly off-center.

![Screen Shot 2022-08-01 at 4 04 04 PM](https://user-images.githubusercontent.com/22647707/182235464-5ed00ca4-72c4-4052-b750-db78ce02da7b.png)

Reply:
Logic was incorporated to clean this up